Add support for WPT style test-wait class name

rniwa opened the pull request:
#### b4b50d4068c80ce1c9db8c90717b72a9046674e1
<pre>
Add support for WPT style test-wait class name
<a href="https://bugs.webkit.org/show_bug.cgi?id=242743">https://bugs.webkit.org/show_bug.cgi?id=242743</a>

Reviewed by Darin Adler and Simon Fraser.

This patch adds the support for test-wait class on the document element to prevent the test from
completion much like <a href="https://commits.webkit.org/229177@main">https://commits.webkit.org/229177@main</a>, which added the support for reftest-wait.

* LayoutTests/fast/harness/test-wait-expected.txt: Added.
* LayoutTests/fast/harness/test-wait.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:

* Tools/DumpRenderTree/DumpRenderTree.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/FrameLoadDelegate.mm:
(-[FrameLoadDelegate dumpAfterWaitAttributeIsRemoved:]):
* Tools/DumpRenderTree/win/FrameLoadDelegate.cpp:
(dumpAfterWaitAttributeIsRemoved):
* Tools/TestRunnerShared/CMakeLists.txt:
* Tools/TestRunnerShared/WPTFunctions.cpp: Renamed from Tools/TestRunnerShared/ReftestFunctions.cpp.
(WTR::sendTestRenderedEvent):
(WTR::hasTestWaitAttribute): Renamed from hasReftestWaitAttribute and added the check for test-wait.
* Tools/TestRunnerShared/WPTFunctions.h: Renamed from Tools/TestRunnerShared/ReftestFunctions.h.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::hasTestWaitAttribute): Renamed from hasReftestWaitAttribute.
(WTR::dumpAfterWaitAttributeIsRemoved):
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/252487@main">https://commits.webkit.org/252487@main</a>
</pre>

darinadler reviewed (approved):
No progressions in WPT from adding this?

smfr commented on LayoutTests/fast/harness/test-wait.html:
```diff
@@ -0,0 +1,17 @@
+    setTimeout(() => {
+        result.textContent = 'PASS';
+        document.documentElement.removeAttribute('class');
+    }, 100);
```
Does this have to be as long as 100ms?. That's a pretty slow test.

smfr replied:
You could do something like have two requestAnimationFrames, remove `test-wait` in the first one, and make sure the second one's log doesn't show in the results.

rniwa replied:
oh, I was just using 100ms as a safe time. I reduced it to 10ms.

rniwa committed:
Add support for WPT style test-wait class name
https://bugs.webkit.org/show_bug.cgi?id=242743

Reviewed by Darin Adler and Simon Fraser.

This patch adds the support for test-wait class on the document element to prevent the test from
completion much like https://commits.webkit.org/229177@main, which added the support for reftest-wait.

* LayoutTests/fast/harness/test-wait-expected.txt: Added.
* LayoutTests/fast/harness/test-wait.html: Added.
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/filter-effects/crashtests/multiple-references-id-crash-002-expected.txt:

* Tools/DumpRenderTree/DumpRenderTree.xcodeproj/project.pbxproj:
* Tools/DumpRenderTree/mac/FrameLoadDelegate.mm:
(-[FrameLoadDelegate dumpAfterWaitAttributeIsRemoved:]):
* Tools/DumpRenderTree/win/FrameLoadDelegate.cpp:
(dumpAfterWaitAttributeIsRemoved):
* Tools/TestRunnerShared/CMakeLists.txt:
* Tools/TestRunnerShared/WPTFunctions.cpp: Renamed from Tools/TestRunnerShared/ReftestFunctions.cpp.
(WTR::sendTestRenderedEvent):
(WTR::hasTestWaitAttribute): Renamed from hasReftestWaitAttribute and added the check for test-wait.
* Tools/TestRunnerShared/WPTFunctions.h: Renamed from Tools/TestRunnerShared/ReftestFunctions.h.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundlePage.cpp:
(WTR::hasTestWaitAttribute): Renamed from hasReftestWaitAttribute.
(WTR::dumpAfterWaitAttributeIsRemoved):
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/252487@main
